gnssanalysis/gn_download.py

```diff
@@ -268,8 +292,13 @@ def check_n_download_url(url, dwndir, filename=False):
         out_f = _Path(dwndir) / filename
         download_url(url, out_f)
 
+    if uncomp:
+        _subprocess.run(["uncompress", f"{out_f}"])
```

Comment:
Overall I understand your use-case for this functionality and so I'm happy to have it merged and used. But I think the current implementation is brittle, has non-intuitive behaviour, (maybe is sketchy form a security perspective?), and only works for certain compression scenarios.
More broadly then that, I think it's really putting too much functionality into a single function and we should handle potential decompression as a second step (with a wrapper running both if we really need a one-liner to call). I get that it mirrors the design of `check_n_download` but I think the decompression option is a design flaw there too. As a first example, we now have two very similar downloading functions with seemingly very similar decompression options that work entirely differently, `check_n_download` only handles `crk.gz` files, decompressing and also doing crx->rnx conversion (potentially even downloading third-party software and compiling it on your system, which is certainly a little surprising for users!), whereas `check_n_download_url` only handles `.Z` files. If the original `check_n_download` function had of been built with less functionality, downloading and decompressing as seperate steps, we would have been able to reuse that decompression functionality here (potentially expanding its capabilities if need be) and wouldn't have two similar but distinct implementations. Ideally we would have the three steps (check file existence (potentially considering compression extension changes), downloading a URI (using FTP or HTTP (or S3?) depending on the URI), and optionally decompressing as separate composable operations, which we could then stick together in a single and flexible function. But that all starts with giving things less functionality not more.

Reply:
Yeah, I think you make some pretty good points, and it's probably best not to introduce this at all.

In the end, my desire for this PR was get a way to download some of the static model files from our s3 bucket and use them when running the PEA (Ginan) using the `ppp_example.yaml` config.

A better way to accomplish this, as you allude to, is not to edit our pretty sub-standard URL download functions. So instead I've gone and introduced a `decompress_file()` function, which I'll use for my purpose, and be the first step in reforming what we've got.

You can check out that function and see if I should change anything, but at the moment it only handles `.gz` and `.tar.gz` which is all that I'll need (I didn't see a good way to introduce the `.Z` decompression using python native stuff, so I've kicked that down the road)

Reply:
I was happy for the existing solution to go in and be used but the changes you've made and the reimplementation are great. I have some further comments on ways the implementation could be tweaked on a second take but this is great as it is and I'm very happy to hit an approve button.